core/Tracker/Request.php

```diff
@@ -921,4 +928,12 @@ private function getVisitorIdAsBinary($idVisitor)
         }
         return false;
     }
+
+    /**
+     * @return bool
+     */
+    public function isPreFightCorsRequest()
+    {
+        return !empty($this->server['HTTP_ACCESS_CONTROL_REQUEST_HEADERS']) || !empty($this->server['HTTP_ACCESS_CONTROL_REQUEST_METHOD']);
```

Comment:
I guess it should be fine to directly use `$_SERVER` here. There is no need to pass the server variables in constructor.